Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+
+namespace BootCamp.Chapter
+{
+    public class Credentials
+    {
+        private const string Separator = ",";
+
+        public string CredentialsFile { get; } = "credentials.db";
+
+        public Credentials()
+        {
+        }
+
+        public Credentials(string credentialsFile)
+        {
+            CredentialsFile = credentialsFile ?? throw new ArgumentNullException(nameof(credentialsFile));
+        }
+
+        private bool FindUser(User user)
+        {
+            string line;
+            bool found = false;
+
+            StreamReader reader;
+            try
+            {
+                reader = new StreamReader(CredentialsFile);
+                while ((line = reader.ReadLine()) != null)
+                {
+                    var isValid = TryParse(line, out User credentials);
+                    if (isValid && user.Equals(credentials))
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                throw new InvalidCredentialsDbFile($"There was an error while trying to work with {CredentialsFile}");
+            }
+            reader?.Close();
+            return found;
+        }
+
+        private bool AddUser(User user)
+        {
+            if (FindUser(user))
+            {
+                throw new UserAllreadyExistsException("User already exists!");
+            }
+            StreamWriter writer;
```

Comment:
not sure if you know about usings? but you should use one! Otherwise if something goes wrong you wont be tidying up resource.

Reply:
I know about using but I chose not to use it but in the process I forgot to add the 'finally' block and close/dispose StreaWriter/Reader.
Check the latest commit [2a4af80](https://github.com/csinn/CSharp-From-Zero-To-Hero/pull/232/commits/2a4af80f11d86f38c8efd0a1debe8b0fd5ae911e)